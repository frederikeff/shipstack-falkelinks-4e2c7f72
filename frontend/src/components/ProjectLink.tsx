'use client';

import Image from "next/image";
import { trackClick } from "@/utils/track";

interface ProjectLinkProps {
  href: string;
  title: string;
  imageSrc: string;
}

export default function ProjectLink({ href, title, imageSrc }: ProjectLinkProps) {
  const handleClick = () => {
    trackClick(href, title);
  };

  return (
    <a
      href={href}
      target="_blank"
      rel="noopener noreferrer"
      className="m-4 flex w-full max-w-2xl items-center rounded-lg bg-white p-4 shadow-lg transition-transform hover:scale-105"
      style={{ color: '#4c2a85' }}
      onClick={handleClick}
    >
      <Image
        src={imageSrc}
        alt={`${title} logo`}
        width={80}
        height={80}
        className="rounded-lg"
      />
      <h2 className="ml-4 text-xl font-semibold">{title}</h2>
    </a>
  );
}