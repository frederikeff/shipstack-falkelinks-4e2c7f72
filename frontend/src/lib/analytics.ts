export const trackEvent = (event: string, data: Record<string, any>) => {
  fetch('/api/track', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ event, ...data }),
  });
};
