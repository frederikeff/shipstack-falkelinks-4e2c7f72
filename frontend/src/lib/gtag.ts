import ReactGA from "react-ga4";

export const initGA = () => {
  const GA_TRACKING_ID = process.env.NEXT_PUBLIC_GA_ID;
  if (GA_TRACKING_ID) {
    ReactGA.initialize(GA_TRACKING_ID);
  }
};

export const logEvent = (category: string, action: string, label: string) => {
  const GA_TRACKING_ID = process.env.NEXT_PUBLIC_GA_ID;
  if (GA_TRACKING_ID) {
    ReactGA.event({
      category,
      action,
      label,
    });
  }
};
