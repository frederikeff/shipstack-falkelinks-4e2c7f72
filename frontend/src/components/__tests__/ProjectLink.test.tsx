import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import ProjectLink from '../ProjectLink';
import * as gtag from '@/lib/gtag';

jest.mock('@/lib/gtag');

describe('ProjectLink', () => {
  it('sends a Google Analytics event on click', async () => {
    const user = userEvent.setup();
    render(
      <ProjectLink
        href="https://www.nxtconnect.ai"
        title="Nxtconnect"
        imageSrc="https://ui-avatars.com/api/?name=NxtConnect&size=400&background=6366f1&color=fff&bold=true"
      />
    );

    const link = screen.getByRole('link');
    await user.click(link);

    expect(gtag.event).toHaveBeenCalledWith({
      action: 'click',
      category: 'Project Link',
      label: 'Nxtconnect',
      value: 0,
    });
  });
});
