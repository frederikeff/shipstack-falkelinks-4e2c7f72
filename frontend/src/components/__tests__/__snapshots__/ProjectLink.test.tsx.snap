// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`renders correctly 1`] = `
<div>
  <a
    class="m-4 flex w-full max-w-2xl items-center rounded-lg bg-white p-4 text-brand-purple shadow-lg transition-transform hover:scale-105"
    href="https://example.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      alt="Example logo"
      class="rounded-lg"
      data-nimg="1"
      decoding="async"
      height="80"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=256&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=96&q=75 1x, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=256&q=75 2x"
      style="color: transparent;"
      width="80"
    />
    <h2
      class="ml-4 text-xl font-semibold"
    >
      Example
    </h2>
  </a>
</div>
`;
